### PR TITLE
Update configuration for opendatahub-operator

### PR DESCRIPTION
**Pull request by ajaypratap003**
Add/update OpenShift CI configuration for opendatahub-opereator.

- Add skip_if_only_changed to avoid CI jobs for non logical changes.
-  Moved build_root base image config in opendata-hub upstream repo and OpenShift CI will pick it from opendatahub-operator repository automatically.
- Add use_build_cache

### CODE REVIEW

**Comment by ajaypratap003** [comment]
cc @VaishnaviHire @zdtsw

File: ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-incubation.yaml [hunk 1]
```diff
@@ -47,7 +46,11 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: opendatahub-operator-image-mirror
   postsubmit: true
+<<<<<<< HEAD
```
Comment (ykaliuta) [inline ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-incubation.yaml:49]: You are committing conflict marks.

Comment (ajaypratap003) [reply ci-operator/config/opendatahub-io/opendatahub-operator/opendatahub-io-opendatahub-operator-incubation.yaml:49]: yes, resolving it. My system kind of hand with thousands of files while syncing with upstream. I've resolved but due to system hang it commit.

**Comment by ajaypratap003** [comment]
Closing this PR and open a fresh PR as there are lot of conflicts issues due to old PR.
https://github.com/openshift/release/pull/52360
fyi @ykaliuta

**Comment by ykaliuta** [comment]
> Closing this PR and open a fresh PR as there are lot of conflicts issues due to old PR. #52360 fyi @ykaliuta

Hmm? Why don't you just update the branch?

**Pull request closed without merging**